src/pds/registry/utils/treks/create_treks_pds4.py

```diff
@@ -0,0 +1,166 @@
+
+    args = parser.parse_args()
+
+    # parse args
+    save_xml = args.save_xml
+    dest = args.destination_directory
+    verbose = args.verbose
+    save_logs = args.save_logs
+    target = args.target.lower()
+
+    # remove whitespace from target
+    target = target.strip()
+    target = target.replace(" ", "")
+
+    selected_targets = []
+    if target == 'all':
+        selected_targets = valid_targets[:-1]
+    else:
+        selected_targets = [target]
+
+    # initialize logger
+    if save_logs:
+        logging.basicConfig(filename='create-treks-pds4-log.log', level=logging.WARNING)
+        logging.warning("target,product_label,missing_data_location,tag")
+
+    total_start = time.time()
+    for target in selected_targets:
+        target_dest = dest + "/" + target
+
+        if verbose:
+            print("Creating pds4 xml for target:", target)
+
+        if save_xml:
+            # create destination path if it does not exist
+            Path(target_dest).mkdir(parents=True, exist_ok=True)
+
+        # get json from url
+        url = "https://trek.nasa.gov/" + target + "/TrekServices/ws/index/eq/" + \
+            "listVisibleLayers?proj=urn:ogc:def:crs:EPSG::60620&start=0&rows=2147483647"
+        response = requests.get(url, timeout=30)
+        json_data = response.json()
+
+        # get data from response
+        # num_found = json_data["response"]["numFound"]
+        start = json_data["response"]["start"]
+        docs = json_data["response"]["docs"]
+
+        # go through each doc and make pds4 xml
+        inventory_entries = []
+        start_time = time.time()
+        for i in range(start, len(docs)):
+
+            data = docs[i]
+
+            # if verbose:
+            label = data["productLabel"]
+            print(f"[{i} / {len(docs) - 1}] | Creating PDS4 label for: {label}")
+
+            psb = ProductServiceBuilder(data=data,
+                                        target=target,
+                                        save_xml=save_xml,
+                                        dest=target_dest,
+                                        verbose=verbose,
+                                        save_logs=save_logs)
+
+            _, lidvid = psb.create_pds4_xml()
+            entry = "P," + lidvid + "\n"
+            inventory_entries.append(entry)
+
+        if verbose:
+            end_time = time.time()
+            total = end_time - start_time
+            print(f"{target.capitalize()} layers took {total:.3f} seconds to build")
+
+        if save_xml:
+            # create collection inventory file
+            with open(target_dest + "/" + target + "_treks_layers_inventory.tab", "w") as f:
+                for entry in inventory_entries:
+                    f.write(entry)
+
+            if verbose:
+                print(f"Collection inventory saved for {target}")
+
+            # create product collection file
+            template_path = os.path.dirname(os.path.abspath(__file__))
```

Comment:
You almost _never_ want to use `__file__`.

If this code is running out of a `.whl` or a `.egg` or any other packages format, you can't predict what `__file__` means (a temporary tree on disk, an in-memory rep, etc.)

Instead, use `importlib` to discover and read non-Python resources that are packaged as a part of the distribution:

```python
import importlib.resources
from pds.registry.utils.treks import templates
…
with importlib.resources.open_text(templates, "product-collection-template.xml") as io:
  template_text = io.read()
…
```
This may require some changes to `create_collection_pds4`.

Reply:
Sounds good, thanks for the input!